README.md
3,000 Alumni Strong
===================

## Created using HTML5 and jQuery 1.9.1 (Initializr)

###[Demo — ver. 1.2.0, ver. 2.0.0, ver. 2.1.0](http://kkirsche.github.io/3-000_Alumni_Strong/)

#### Version 2.2.0
![version 2.2.0 - 3,000 Alumni Strong](http://i.imgur.com/rhNm1no.jpg "3,000 Alumni Strong")
[version 2.1.0 - 3,000 Alumni Strong](http://i.imgur.com/4lTMYyo.jpg "3,000 Alumni Strong")<br />
[version 2.0.0 - 3,000 Alumni Strong](http://i.imgur.com/CIoDykC.jpg "3,000 Alumni Strong")<br />
[version 1.3.0 - 3,000 Alumni Strong](http://i.imgur.com/z0hHNpY.jpg "3,000 Alumni Strong")<br />
[version 1.1.0 - 3,000 Alumni Strong](http://i.imgur.com/JcG4qBI.jpg "3,000 Alumni Strong")<br />
[version 1.0.0 - 3,000 Alumni Strong](http://i.imgur.com/ldtkprf.jpg "3,000 Alumni Strong")<br />
[version 0.5.0 - 3,000 Alumni Strong](http://i.imgur.com/ycPNk67.jpg "3,000 Alumni Strong")<br />
[version 0.4.0 - 3,000 Alumni Strong](http://i.imgur.com/0FnBthK.jpg "3,000 Alumni Strong")<br />
[version 0.3.0 - 3,000 Alumni Strong](http://i.imgur.com/W1sdlmw.jpg "3,000 Alumni Strong")<br />
[version 0.2.0 — 3,000 Alumni Strong](http://i.imgur.com/SrrG093.jpg "3,000 Alumni Strong")<br />
[version 0.1.0 — 3,000 Alumni Strong](http://i.imgur.com/ctigLrQ.jpg "3,000 Alumni Strong")

## Versioning

For transparency and insight into our release cycle, and for striving to maintain backward compatibility, 3,000 Alumni Strong will be maintained under the Semantic Versioning guidelines as much as possible.

Releases will be numbered with the following format:

`<major>.<minor>.<patch>`

And constructed with the following guidelines:

* Breaking backward compatibility bumps the major (and resets the minor and patch)
* New additions without breaking backward compatibility bumps the minor (and resets the patch)
* Bug fixes and misc changes bumps the patch

For more information on SemVer, please visit [http://semver.org/](http://semver.org/).

CHANGELOG.md
## Changelog

### Version 2.2.0
Files Changed:
* index.html
* js/main.js

Changes made:
* Change text per Spencer's request
* Add new video

### Version 2.1.1
Files Changed:
* index.html
* js/main.js

Changes made:
* Fix type ```&rsqup;``` with ```&rsquo;```
* Fix typos in paragraph

### Version 2.1.0
#### Third Video
Files Changed:
* index.html
* /img/3000StrongLogo300.png
* js/main.js
* README.md
* /img/AlumniDonorProgress-mailer2.png

Changes made:
* Updated copy to the left of the video
* Updated video carousel to load new title text
* Updated demo to include mailer 3 versions
* Updated Alumni Donors progress graph

### Version 2.0.0
#### Second Mailer
Files Changed:
* index.html
* /img/carousel/*
* js/main.js
* README.md
* /img/AlumniDonorProgress-mailer2.png

Changes made:
* Updated carousel copy
* Updated video carousel to load new title text
* Updated carousel images
* Updated demo to include mailer 1 and mailer 2 versions
* Updated Alumni Donors progress graph

Backwards Compatability Break:
* Images in /img/carousel/ now grouped by which mailer they were in.

### Version 1.3.0

Files changed:
* index.html
* js/main.js
* classProgress.html (removed)
* /img/

Changes Made:
* Add next / previous video functionality
* Removed unnecessary images

<hr />

### Version 1.2.0

Files changed:
* index.html
* css/main.css

Changes Made:
* Major rewrite of the top menu bar to work with mobile (custom to bootstrap's navbar)
* Restructuring of code on the page to better scale to mobile devices
* Updated media queries to reflect changes to mobile devices.

<hr />

### Version 1.1.0

Files changed:
* index.html
* css/main.css

Changes Made:
* Text changes in index.html
* Red headers now handled via CSS classes instead of for all headers
* Carousel javascript updated
* Donor Number can be updated automatically via javascript
* Modest YouTube Branding
* Show video information off by default (video title, etc. at the top on load).

<hr />

### Version 1.0.0
#### First Mailer Release

js/main-mailer3.js
$(document).ready(function () {
    "use strict";
    //Variables
    var whichVideoIsNext = 2;
    function switchVideos(videoNumber) {
        var newVideoURL = "", newVideoTitle = "";
        $(".videoWrapper").fadeOut("fast");
        switch (videoNumber) {
            case 1:
                newVideoTitle = "General Sam&rsquo;s Story for the 3,000 Alumni Strong Challenge";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/_gRmUAqaCwk?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            case 2:
                newVideoTitle = "Life on Campus: Spring Semester 2013";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/XK93kVGLZ6Y?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            case 3:
                newVideoTitle = "Dr. Pontuso supports 3,000 Strong Challenge";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/eYhsyjCU5-s?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            default:
                newVideoTitle = "";
                newVideoURL = "There was a problem switching videos.";
                break;
        }
        $("#videoTitle").html(newVideoTitle);
        $(".videoWrapper").html(newVideoURL);
        $(".videoWrapper").fadeIn("fast");
        return false;
    }
    $("[name=nextVideo]").click(function(e) {
        e.preventDefault();
        if (whichVideoIsNext > 3) {
            whichVideoIsNext = 1;
        } else if (whichVideoIsNext < 1) {
            whichVideoIsNext = 1;
        }
        switchVideos(whichVideoIsNext);
        whichVideoIsNext++;
    });
    $("[name=prevVideo]").click(function(e) {
        e.preventDefault();
        whichVideoIsNext -= 2;
        if (whichVideoIsNext > 3) {
            whichVideoIsNext = 3;
        } else if (whichVideoIsNext < 1) {
            whichVideoIsNext = 3;
        }
        switchVideos(whichVideoIsNext);
        whichVideoIsNext++;
    });
});


Process/ORIGINAL DESIGN/js/decadeSwitchingLogic.js
function calculatePercentage(numberOfDonors, TotalNumberOfAlumni) {
    "use strict";
    //create variable and set a value. This is to ensure we don't return nil.
    var percentOfAlumni = 0;
    if (numberOfDonors < 0) {
        return 0;
    } else if (TotalNumberOfAlumni <= 0) {
        return 0;
    }

    //Donors / Alumni gives us decimal percent, multiply by 100 for numerical percentage.
    percentOfAlumni = (100 * (numberOfDonors / TotalNumberOfAlumni));

    if (percentOfAlumni > 100) {
        percentOfAlumni = 100;
    } else if (percentOfAlumni < 0) {
        percentOfAlumni = 0;
    }

    return percentOfAlumni;
}

function showHiddenBars() {
    "use strict";
    //show the bars we aren't using.
    $('#decade10').css("display", "inline-block");
    $('#decade11').css("display", "inline-block");
    $('#decade12').css("display", "inline-block");
    $('#decade13').css("display", "inline-block");
    $('#decade10label').css("display", "inline");
    $('#decade11label').css("display", "inline");
    $('#decade12label').css("display", "inline");
    $('#decade13label').css("display", "inline");
    $('#decade10AmountGiven').css("display", "inline");
    $('#decade11AmountGiven').css("display", "inline");
    $('#decade12AmountGiven').css("display", "inline");
    $('#decade13AmountGiven').css("display", "inline");
}
function hideHiddenBars() {
    "use strict";
    //hide the bars we aren't using.
    $('#decade10').css("display", "none");
    $('#decade11').css("display", "none");
    $('#decade12').css("display", "none");
    $('#decade13').css("display", "none");
    $('#decade10label').css("display", "none");
    $('#decade11label').css("display", "none");
    $('#decade12label').css("display", "none");
    $('#decade13label').css("display", "none");
    $('#decade10AmountGiven').css("display", "none");
    $('#decade11AmountGiven').css("display", "none");
    $('#decade12AmountGiven').css("display", "none");
    $('#decade13AmountGiven').css("display", "none");
}

function initilizeBars2000to2012(decade1Donors, decade1TotalDonors, decade2Donors, decade2TotalDonors, decade3Donors, decade3TotalDonors, decade4Donors, decade4TotalDonors, decade5Donors, decade5TotalDonors, decade6Donors, decade6TotalDonors, decade7Donors, decade7TotalDonors, decade8Donors, decade8TotalDonors, decade9Donors, decade9TotalDonors, decade10Donors, decade10TotalDonors, decade11Donors, decade11TotalDonors, decade12donors, decade12TotalDonors, decade13donors, decade13TotalDonors) {
    "use strict";
    var decade1Percentage = 0, decade2Percentage = 0, decade3Percentage = 0, decade4Percentage = 0, decade5Percentage = 0, decade6Percentage = 0, decade7Percentage = 0, decade8Percentage = 0, decade9Percentage = 0, decade10Percentage = 0, decade11Percentage = 0, decade12Percentage = 0, decade13Percentage = 0;

    decade1Percentage = calculatePercentage(decade1Donors, decade1TotalDonors);
    decade2Percentage = calculatePercentage(decade2Donors, decade2TotalDonors);
    decade3Percentage = calculatePercentage(decade3Donors, decade3TotalDonors);
    decade4Percentage = calculatePercentage(decade4Donors, decade4TotalDonors);
    decade5Percentage = calculatePercentage(decade5Donors, decade5TotalDonors);
    decade6Percentage = calculatePercentage(decade6Donors, decade6TotalDonors);
    decade7Percentage = calculatePercentage(decade7Donors, decade7TotalDonors);
    decade8Percentage = calculatePercentage(decade8Donors, decade8TotalDonors);
    decade9Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);
    decade10Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);
    decade11Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);
    decade12Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);
    decade13Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);

    //update the progress bar with our calculated percentage
    $("#decade1").progressbar({
        value: decade1Percentage
    });
    $("#decade2").progressbar({
        value: decade2Percentage
    });
    $("#decade3").progressbar({
        value: decade3Percentage
    });
    $("#decade4").progressbar({
        value: decade4Percentage
    });
    $("#decade5").progressbar({
        value: decade5Percentage
    });
    $("#decade6").progressbar({
        value: decade6Percentage
    });
    $("#decade7").progressbar({
        value: decade7Percentage
    });
    $("#decade8").progressbar({
        value: decade8Percentage
    });
    $("#decade9").progressbar({
        value: decade9Percentage
    });
    $("#decade10").progressbar({
        value: decade10Percentage
    });
    $("#decade11").progressbar({
        value: decade11Percentage
    });
    $("#decade12").progressbar({
        value: decade12Percentage
    });
    $("#decade13").progressbar({
        value: decade13Percentage
    });
}
//Updates the 2010-Present Progress bars, hides the ones that aren't used.
function updateBars12Years(decade1Donors, decade1TotalDonors, decade2Donors, decade2TotalDonors, decade3Donors, decade3TotalDonors, decade4Donors, decade4TotalDonors, decade5Donors, decade5TotalDonors, decade6Donors, decade6TotalDonors, decade7Donors, decade7TotalDonors, decade8Donors, decade8TotalDonors, decade9Donors, decade9TotalDonors, decade10Donors, decade10TotalDonors, decade11Donors, decade11TotalDonors, decade12Donors, decade12TotalDonors, decade13Donors, decade13TotalDonors) {
    "use strict";
    var decade1Percentage = 0, decade2Percentage = 0, decade3Percentage = 0, decade4Percentage = 0, decade5Percentage = 0, decade6Percentage = 0, decade7Percentage = 0, decade8Percentage = 0, decade9Percentage = 0, decade10Percentage = 0, decade11Percentage = 0, decade12Percentage = 0, decade13Percentage = 0;

    //Calculate the percentages
    decade1Percentage = calculatePercentage(decade1Donors, decade1TotalDonors);
    decade2Percentage = calculatePercentage(decade2Donors, decade2TotalDonors);
    decade3Percentage = calculatePercentage(decade3Donors, decade3TotalDonors);
    decade4Percentage = calculatePercentage(decade4Donors, decade4TotalDonors);
    decade5Percentage = calculatePercentage(decade5Donors, decade5TotalDonors);
    decade6Percentage = calculatePercentage(decade6Donors, decade6TotalDonors);
    decade7Percentage = calculatePercentage(decade7Donors, decade7TotalDonors);
    decade8Percentage = calculatePercentage(decade8Donors, decade8TotalDonors);
    decade9Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);
    decade10Percentage = calculatePercentage(decade10Donors, decade10TotalDonors);
    decade11Percentage = calculatePercentage(decade11Donors, decade11TotalDonors);
    decade12Percentage = calculatePercentage(decade12Donors, decade12TotalDonors);
    decade13Percentage = calculatePercentage(decade13Donors, decade13TotalDonors);

    //Hide bars we aren't using
    showHiddenBars();

    //Update existing progress bars.
    $('#decade1').progressbar("value", decade1Percentage);
    $('#decade2').progressbar("value", decade2Percentage);
    $('#decade3').progressbar("value", decade3Percentage);
    $('#decade4').progressbar("value", decade4Percentage);
    $('#decade5').progressbar("value", decade5Percentage);
    $('#decade6').progressbar("value", decade6Percentage);
    $('#decade7').progressbar("value", decade7Percentage);
    $('#decade8').progressbar("value", decade8Percentage);
    $('#decade9').progressbar("value", decade9Percentage);
    $('#decade10').progressbar("value", decade10Percentage);
    $('#decade11').progressbar("value", decade11Percentage);
    $('#decade12').progressbar("value", decade12Percentage);
    $('#decade13').progressbar("value", decade12Percentage);
}

function updateBars9Years(decade1Donors, decade1TotalDonors, decade2Donors, decade2TotalDonors, decade3Donors, decade3TotalDonors, decade4Donors, decade4TotalDonors, decade5Donors, decade5TotalDonors, decade6Donors, decade6TotalDonors, decade7Donors, decade7TotalDonors, decade8Donors, decade8TotalDonors, decade9Donors, decade9TotalDonors) {
    "use strict";
    var decade1Percentage = 0, decade2Percentage = 0, decade3Percentage = 0, decade4Percentage = 0, decade5Percentage = 0, decade6Percentage = 0, decade7Percentage = 0, decade8Percentage = 0, decade9Percentage = 0;

    //Calculate the percentages
    decade1Percentage = calculatePercentage(decade1Donors, decade1TotalDonors);
    decade2Percentage = calculatePercentage(decade2Donors, decade2TotalDonors);
    decade3Percentage = calculatePercentage(decade3Donors, decade3TotalDonors);
    decade4Percentage = calculatePercentage(decade4Donors, decade4TotalDonors);
    decade5Percentage = calculatePercentage(decade5Donors, decade5TotalDonors);
    decade6Percentage = calculatePercentage(decade6Donors, decade6TotalDonors);
    decade7Percentage = calculatePercentage(decade7Donors, decade7TotalDonors);
    decade8Percentage = calculatePercentage(decade8Donors, decade8TotalDonors);
    decade9Percentage = calculatePercentage(decade9Donors, decade9TotalDonors);

    //Hide bars we aren't using
    hideHiddenBars();

    //Update existing progress bars.
    $('#decade1').progressbar("value", decade1Percentage);
    $('#decade2').progressbar("value", decade2Percentage);
    $('#decade3').progressbar("value", decade3Percentage);
    $('#decade4').progressbar("value", decade4Percentage);
    $('#decade5').progressbar("value", decade5Percentage);
    $('#decade6').progressbar("value", decade6Percentage);
    $('#decade7').progressbar("value", decade7Percentage);
    $('#decade8').progressbar("value", decade8Percentage);
    $('#decade9').progressbar("value", decade9Percentage);
}

js/main.js
$(document).ready(function () {
    "use strict";
    //Variables
    var whichVideoIsNext = 2;
    function switchVideos(videoNumber) {
        var newVideoURL = "", newVideoTitle = "";
        $(".videoWrapper").fadeOut("fast");
        switch (videoNumber) {
            case 1:
                newVideoTitle = "&ldquo;Fishin&rsquo; Blues&rdquo; by Professors Cabas and Townsend";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/MQKjobmu-0Y?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            case 2:
                newVideoTitle = "General Sam Shares a Story";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/_gRmUAqaCwk?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            case 3:
                newVideoTitle = "Life on Campus: Spring Semester 2013";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/XK93kVGLZ6Y?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            case 4:
                newVideoTitle = "Dr. Pontuso supports 3,000 Strong Challenge";
                newVideoURL = "<iframe width=\"521\" height=\"293\" src=\"http://www.youtube.com/embed/eYhsyjCU5-s?rel=0&amp;modestbranding=1&amp;showinfo=0\" frameborder=\"0\" allowfullscreen></iframe>";
                break;
            default:
                newVideoTitle = "";
                newVideoURL = "There was a problem switching videos.";
                break;
        }
        $("#videoTitle").html(newVideoTitle);
        $(".videoWrapper").html(newVideoURL);
        $(".videoWrapper").fadeIn("fast");
        return false;
    }
    $("[name=nextVideo]").click(function(e) {
        e.preventDefault();
        if (whichVideoIsNext > 4) {
            whichVideoIsNext = 1;
        } else if (whichVideoIsNext < 1) {
            whichVideoIsNext = 1;
        }
        switchVideos(whichVideoIsNext);
        whichVideoIsNext++;
    });
    $("[name=prevVideo]").click(function(e) {
        e.preventDefault();
        whichVideoIsNext -= 2;
        if (whichVideoIsNext > 4) {
            whichVideoIsNext = 4;
        } else if (whichVideoIsNext < 1) {
            whichVideoIsNext = 4;
        }
        switchVideos(whichVideoIsNext);
        whichVideoIsNext++;
    });
});
